### PR TITLE
install-script: Follow FHS rules

### PR DESCRIPTION
**Pull request by pymumu**
Follow FHS rules
https://github.com/pymumu/smartdns/issues/390

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+#
+# smartdns is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+.PHONY: all
```
Comment (jtojnar) [inline Makefile]: `SMARTDNS_BIN`, `clean` and `install` need to be phony too, otherwise they will only run a single time.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+# (at your option) any later version.
+#
+# smartdns is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
```
Comment (jtojnar) [inline Makefile]:
```suggestion

PKG_CONFIG := pkg-config

DESTDIR :=
PREFIX := /usr
SBINDIR := $(PREFIX)/sbin
SYSCONFDIR := /etc
SYSTEMDSERVICEDIR := $(shell $PKG_CONFIG --variable=systemdsystemunitdir systemd)
```

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+clean:
+	make -C src clean $(MAKEFLAGS)
+
+install: SMARTDNS_BIN
+	install -v -m 0755 -d $(DESTDIR)/etc/smartdns
+	install -v -m 0640 -t $(DESTDIR)/etc/default etc/default/smartdns
+	install -v -m 0755 -t $(DESTDIR)/etc/init.d etc/init.d/smartdns 
+	install -v -m 0640 -t $(DESTDIR)/etc/smartdns etc/smartdns/smartdns.conf
+	install -v -m 0755 -t $(DESTDIR)/usr/sbin src/smartdns
+	install -v -m 0644 -t $(DESTDIR)/lib/systemd/system systemd/smartdns.service
```
Comment (jtojnar) [inline Makefile]:
```suggestion
	install -v -m 0755 -d $(DESTDIR)$(SYSCONFDIR)/smartdns
	install -v -m 0640 -t $(DESTDIR)$(SYSCONFDIR)/default etc/default/smartdns
	install -v -m 0755 -t $(DESTDIR)$(SYSCONFDIR)/init.d etc/init.d/smartdns 
	install -v -m 0640 -t $(DESTDIR)$(SYSCONFDIR)/smartdns etc/smartdns/smartdns.conf
	install -v -m 0755 -t $(DESTDIR)$(SBINDIR) src/smartdns
	install -v -m 0644 -t $(DESTDIR)$(SYSTEMDSYSTEMUNITDIR) systemd/smartdns.service 
```

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+.PHONY: all
+
+all: SMARTDNS_BIN
+
+SMARTDNS_BIN : 
+	make -C src all $(MAKEFLAGS)
+
+clean:
+	make -C src clean $(MAKEFLAGS)
+
```
Comment (jtojnar) [inline Makefile]: I would suggest renaming the service file to `service.in`, replace the hardcoded paths with placeholders and add the following target to the makefile:

```suggestion

systemd/smartdns.service: systemd/smartdns.service.in
    sed 's|@SBINDIR@|$(SBINDIR)|' $< > $@
    sed 's|@SYSCONFDIR@|$(SYSCONFDIR)|' $< > $@
    sed 's|@RUNSTATEDIR@|$(RUNSTATEDIR)|' $< > $@
```

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+.PHONY: all
+
+all: SMARTDNS_BIN
+
+SMARTDNS_BIN : 
+	make -C src all $(MAKEFLAGS)
```
Comment (jtojnar) [inline Makefile]: When using [recursive make](https://www.gnu.org/software/make/manual/html_node/Recursion.html), you should use MAKE variable:

```suggestion
	$(MAKE) -C src all $(MAKEFLAGS)
```

Though note recursive make is deemed problematic by some (e.g. https://embeddedartistry.com/blog/2017/04/10/recursive-make-considered-harmful/). But for a small project you can probably get away with it.

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+	cp $< $@
+	sed -i 's|@SBINDIR@|$(SBINDIR)|' $@
+	sed -i 's|@SYSCONFDIR@|$(SYSCONFDIR)|' $@
+	sed -i 's|@RUNSTATEDIR@|$(RUNSTATEDIR)|' $@
+
+clean:
+	$(MAKE) -C src clean $(MFLAGS) 
+	$(RM) $(SMARTDNS_SYSTEMD)
+
+install: SMARTDNS_BIN
```
Comment (jtojnar) [inline Makefile]:
```suggestion
install: SMARTDNS_BIN $(SMARTDNS_SYSTEMD)
```

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+	$(MAKE) -C src clean $(MFLAGS) 
+	$(RM) $(SMARTDNS_SYSTEMD)
+
+install: SMARTDNS_BIN 
+	install -v -m 0755 -d $(DESTDIR)$(SYSCONFDIR)/smartdns
+	install -v -m 0640 -t $(DESTDIR)$(SYSCONFDIR)/default etc/default/smartdns
+	install -v -m 0755 -t $(DESTDIR)$(SYSCONFDIR)/init.d etc/init.d/smartdns 
+	install -v -m 0640 -t $(DESTDIR)$(SYSCONFDIR)/smartdns etc/smartdns/smartdns.conf
+	install -v -m 0755 -t $(DESTDIR)$(SBINDIR) src/smartdns
+	install -v -m 0644 -t $(DESTDIR)$(SYSTEMDSYSTEMUNITDIR) systemd/smartdns.service
```
Comment (jtojnar) [inline Makefile]: `-D` should create the directories if they do not exist:

```suggestion
	install -v -m 0640 -D -t $(DESTDIR)$(SYSCONFDIR)/default etc/default/smartdns
	install -v -m 0755 -D -t $(DESTDIR)$(SYSCONFDIR)/init.d etc/init.d/smartdns
	install -v -m 0640 -D -t $(DESTDIR)$(SYSCONFDIR)/smartdns etc/smartdns/smartdns.conf
	install -v -m 0755 -D -t $(DESTDIR)$(SBINDIR) src/smartdns
	install -v -m 0644 -D -t $(DESTDIR)$(SYSTEMDSYSTEMUNITDIR) systemd/smartdns.service
```

File: Makefile (hunk 6 above)
Comment (pymumu) [reply Makefile]: the dependence is at line 28
```
SMARTDNS_BIN: $(SMARTDNS_SYSTEMD)
```

Comment (jtojnar) [reply Makefile]: You are right. Your location is also better.

**Comment by pymumu** [comment]
Thank you very much for your guidance.

**Pull request merged**